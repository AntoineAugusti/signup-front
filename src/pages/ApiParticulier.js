import React from 'react';
import PropTypes from 'prop-types';
import Form from '../components/Form';
import Nav from '../components/Nav';

// Description du contexte
const target_api = 'api_particulier';
const title = "Demande d'accès à API Particulier";

// Le demandeur doit décrire le contexte d'usage de l'API
const DemarcheDescription = () => (
  <div className="text-quote">
    <p>
      Pour avoir accès à l&apos;API Particulier, diffusant des données
      personnelles, vous devez obtenir un agrément. L&apos;accès à cette API
      n&apos;est pour l&apos;instant disponible que si vous êtes:
    </p>
    <ul>
      <li>une administration</li>
      <li>
        une entreprise prestataire d&apos;une administration ou ayant une
        délégation de service public
      </li>
    </ul>
    <p>
      Pour utiliser API Particulier, vous devez vous engager à traiter la bonne
      donnée par le bon agent de votre administration et informer correctement
      l’usager.
    </p>
  </div>
);
const isFranceConnected = false;

const contacts = {
  technique: {
    heading: 'Responsable technique',
    description: () => (
      <p>
        Cette personne recevra les accès techniques par mail. Le numéro de
        téléphone doit être un numéro de téléphone mobile. Il sera utilisé pour
        envoyer un code d'accès. Le responsable technique peut être le contact
        technique de votre prestataire.
      </p>
    ),
    email: '',
    phone_number: '',
  },
};

const CadreJuridiqueDescription = () => (
  <div className="text-quote">
    <p>
      Pour pouvoir bénéficier du raccordement à l&lsquo;API Particulier, le
      cadre légal et réglementaire des fournisseurs de service doit permettre à
      la DINSIC de transmettre des données personnelles à votre entité
      administrative.
    </p>
  </div>
);

// Le demandeur doit séléctionner les données auxquelles il demande l'accès
const DonneesDescription = () => (
  <div className="text-quote">
    <p>
      La loi informatique et libertés définit les principes à respecter lors de
      la collecte, du traitement et de la conservation de données personnelles.
    </p>
    <p>L’article 6 précise :</p>
    <ul>
      <li>
        3° [les données] sont adéquates, pertinentes et non excessives au regard
        des finalités pour lesquelles elles sont collectées et de leurs
        traitements ultérieurs ;
      </li>
      <li>
        4° Elles sont exactes, complètes et, si nécessaire, mises à jour ; les
        mesures appropriées doivent être prises pour que les données inexactes
        ou incomplètes au regard des finalités pour lesquelles elles sont
        collectées ou traitées soient effacées ou rectifiées ;
      </li>
    </ul>
    <p>
      Nous vous remercions de sélectionner uniquement les données strictement
      nécessaires à votre téléservice. Le non-respect du principe de
      proportionnalité vous expose vis à vis de la CNIL.
    </p>
  </div>
);

const availableScopes = [
  {
    name: 'dgfip_avis_imposition',
    humanName: "DGFIP - Avis d'imposition",
  },
  {
    name: 'dgfip_adresse',
    humanName: 'DGFIP - Adresse',
  },
  {
    name: 'cnaf_quotient_familial',
    humanName: 'CNAF - Quotient familial',
  },
  {
    name: 'cnaf_attestation_droits',
    humanName: 'CNAF - Attestation de droits',
  },
];

// Le demandeur valide les modalités d'utilisation
const CguDescription = () => <React.Fragment />;
const cguLink = 'https://particulier.api.gouv.fr/API_Particulier_modalites.pdf';

const ApiParticulier = ({
  match: {
    params: { enrollmentId },
  },
}) => (
  <div className="dashboard">
    <Nav
      navLinksGeneral={[
        { id: 'description', label: 'Description' },
        { id: 'donnees', label: 'Données' },
        { id: 'cadre-juridique', label: 'Cadre juridique' },
        { id: 'donnees-personnelles', label: 'Données personnelles' },
        { id: 'contacts-moe', label: 'Mise en œuvre' },
        { id: 'cgu', label: "Modalités d'utilisation" },
      ]}
      contactInformation={[
        {
          email: 'contact@api.gouv.fr',
          label: 'Nous contacter',
          subject: 'Contact%20via%20signup.api.gouv.fr%20-%20API%20Particulier',
        },
      ]}
    />
    <div className="main">
      <Form
        enrollmentId={enrollmentId}
        target_api={target_api}
        title={title}
        DemarcheDescription={DemarcheDescription}
        isFranceConnected={isFranceConnected}
        contacts={contacts}
        CadreJuridiqueDescription={CadreJuridiqueDescription}
        DonneesDescription={DonneesDescription}
        availableScopes={availableScopes}
        CguDescription={CguDescription}
        cguLink={cguLink}
      />
    </div>
  </div>
);

ApiParticulier.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      enrollmentId: PropTypes.string,
    }),
  }),
};

ApiParticulier.defaultProps = {
  match: {
    params: {
      enrollmentId: null,
    },
  },
};

export default ApiParticulier;
