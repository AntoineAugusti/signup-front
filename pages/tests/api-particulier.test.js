import ReactTestRenderer from 'react-test-renderer'
import ApiParticulier from '../../pages/api-particulier'

require('../../lib/tests/local-storage') // eslint-disable-line import/no-unassigned-import

describe('pages | ApiParticulier', () => {
  describe('render', () => {
    const renderer = ReactTestRenderer.create(<ApiParticulier />)
    it('should be defined', () => {
      expect(renderer).toBeDefined()
    })
    it('should match snapshot', () => {
      expect(renderer).toMatchSnapshot()
    })
  })
})
