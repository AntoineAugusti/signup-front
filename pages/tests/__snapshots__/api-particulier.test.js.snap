// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages | ApiParticulier render should match snapshot 1`] = `
<div
  className="page"
>
  <header>
    <nav
      className="nav"
    >
      <div
        className="nav__container"
      >
        <a
          className="nav__link"
          href="https://api.gouv.fr/"
        >
          <img
            alt="Accueil de signup.api.gouv.fr"
            className="nav__logo"
            src="https://api.gouv.fr/img/logo_api.svg"
          />
        </a>
        <ul
          className="nav__links"
        >
          <li>
            <a
              href="http://localhost:3022/oauth/authorize?client_id=82546188522214c3577d35c283ce8842786649b35a026a9d44908037a597f29b&redirect_uri=http%3A%2F%2Flocalhost%3A3000&scope=&response_type=token&state="
            >
              Se connecter
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div>
    
    <div
      className="documentation"
    >
      <nav
        className="side-pane"
      >
        <ul
          className="side-pane__menu"
        >
          <li
            className="side-pane__title"
          >
            <h3>
              Votre demande
            </h3>
          </li>
          <li>
            <a
              className="side-pane__link"
              href="#legal"
            >
              Fondement juridique
            </a>
          </li>
          <li>
            <a
              className="side-pane__link"
              href="#donnees"
            >
              Choix des données
            </a>
          </li>
          <li>
            <a
              className="side-pane__link"
              href="#volumetrie"
            >
              Volumétrie
            </a>
          </li>
          <li>
            <a
              className="side-pane__link"
              href="#cnil"
            >
              Obligation CNIL
            </a>
          </li>
          <li>
            <a
              className="side-pane__link"
              href="#convention"
            >
              Convention
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="main-pane"
      >
        <form
          onSubmit={[Function]}
        >
          <h1
            id="legal"
          >
            Fondement juridique
          </h1>
          <section
            className="information-text"
          >
            <p>
              Pour pouvoir bénéficier du raccordement à l‘API Particulier, le cadre légal et réglementaire des fournisseurs de service doit permettre à la DINSIC de leur transmettre des données à caractère personnel.
            </p>
          </section>
          <div
            className="form__group"
          >
            <label
              htmlFor="fournisseur_de_service"
            >
              Nom du fournisseur de service
            </label>
            <input
              id="fournisseur_de_service"
              name="enrollment.fournisseur_de_service"
              onChange={[Function]}
              type="text"
              value={undefined}
            />
          </div>
          <div
            className="form__group"
          >
            <label
              htmlFor="description_service"
            >
              Décrivez brièvement votre service ainsi que l‘utilisation prévue des données transmises
            </label>
            <textarea
              id="description_service"
              name="enrollment.description_service"
              onChange={[Function]}
              value={undefined}
            />
          </div>
          <div
            className="form__group"
          >
            <label
              htmlFor="fondement_juridique"
            >
              Veuillez transmettre le fondement juridique sur lequel s’appuie votre demande (décret, délibération …)
            </label>
            <textarea
              id="fondement_juridique"
              name="enrollment.fondement_juridique"
              onChange={[Function]}
              value={undefined}
            />
          </div>
          <h1
            id="donnees"
          >
            Choix des données
          </h1>
          <section
            className="information-text"
          >
            <p>
              Les règles relatives à la protection de la vie privée fixent les principes à respecter lors de la collecte, du traitement et de la conservation de données personnelles. Les données ainsi récoltées doivent être adéquates, pertinentes et proportionnelles par rapport au traitement.
            </p>
            <p>
              Nous vous remercions de sélectionner uniquement les données strictement nécessaires à votre téléservice. Le non-respect de ces règles vous expose vis à vis de la CNIL.
            </p>
          </section>
          <div
            className="form__group"
          >
            <fieldset
              className="vertical"
            >
              <legend>
                Sélectionnez vos jeux de données souhaités
              </legend>
              <div>
                <input
                  checked={undefined}
                  id="checkbox-scope_dgfip_avis_imposition"
                  name="enrollment.scope_dgfip_avis_imposition"
                  onChange={[Function]}
                  type="checkbox"
                  value="true"
                />
                <label
                  className="label-inline"
                  htmlFor="checkbox-scope_dgfip_avis_imposition"
                >
                  DGFiP - Avis Imposition
                </label>
              </div>
              <div>
                <input
                  checked={undefined}
                  id="checkbox-scope_cnaf_attestation_droits"
                  name="enrollment.scope_cnaf_attestation_droits"
                  onChange={[Function]}
                  type="checkbox"
                  value="scope_cnaf_attestation_droits"
                />
                <label
                  className="label-inline"
                  htmlFor="checkbox-scope_cnaf_attestation_droits"
                >
                  CNAF - Attestation de droits
                </label>
              </div>
              <div>
                <input
                  checked={undefined}
                  id="checkbox-scope_cnaf_quotient_familial"
                  name="enrollment.scope_cnaf_quotient_familial"
                  onChange={[Function]}
                  type="checkbox"
                  value="true"
                />
                <label
                  className="label-inline"
                  htmlFor="checkbox-scope_cnaf_quotient_familial"
                >
                  CNAF - Quotient Familial
                </label>
              </div>
            </fieldset>
          </div>
          <h1
            id="volumetrie"
          >
            Volumétrie
          </h1>
          <section
            className="information-text"
          >
            <p>
              Avoir une information de la volumétrie et, le cas échéant à la saisonnalité de votre démarche, nous permettra de vous offrir la meilleure qualité de service possible en lien avec les différents fournisseurs de vos données.
            </p>
            <p>
              Conformément à nos 
              <a
                href="#convention"
              >
                conditions d’utilisation
              </a>
              , nous nous réservons le droit de réduire ou couper vos appels.
            </p>
          </section>
          <div
            className="form__group"
          >
            <label
              htmlFor="nombre_demandes_annuelle"
            >
              Connaissez-vous le volume global annuel des demandes de votre téléservice ?
            </label>
            <input
              id="nombre_demandes_annuelle"
              name="enrollment.nombre_demandes_annuelle"
              onChange={[Function]}
              type="text"
              value={undefined}
            />
          </div>
          <div
            className="form__group"
          >
            <label
              htmlFor="pic_demandes_par_heure"
            >
              Connaissez-vous le pic de charge (en nombre de demandes horaires) ?
            </label>
            <input
              id="pic_demandes_par_heure"
              name="enrollment.pic_demandes_par_heure"
              onChange={[Function]}
              type="text"
              value={undefined}
            />
          </div>
          <div
            className="form__group"
          >
            <label
              htmlFor="pic_demandes_par_heure"
            >
              Connaissez-vous la répartition de la charge des demandes mensuelles (0 si le service est fermé) ?
            </label>
            <pre>
              <code>
                ‘nombre_demandes_mensuelles_jan‘: 45, ‘nombre_demandes_mensuelles_fev‘: 45, ‘nombre_demandes_mensuelles_mar‘: 45, ‘nombre_demandes_mensuelles_avr‘: 45, ‘nombre_demandes_mensuelles_mai‘: 45, ‘nombre_demandes_mensuelles_jui‘: 45, ‘nombre_demandes_mensuelles_jul‘: 45, ‘nombre_demandes_mensuelles_aou‘: 45, ‘nombre_demandes_mensuelles_sep‘: 45, ‘nombre_demandes_mensuelles_oct‘: 45, ‘nombre_demandes_mensuelles_nov‘: 45, ‘nombre_demandes_mensuelles_dec‘: 45,
              </code>
            </pre>
          </div>
          <h1
            id="cnil"
          >
            Obligation CNIL
          </h1>
          <section
            className="information-text"
          >
            <p>
              L’entrée en vigueur le 25 mai 2018 du règlement européen sur la protection des données supprime totalement le régime déclaratif.
            </p>
            <p>
              Les fournisseurs de services ont en particulier l’obligation de tenir un registre de leurs activités de traitement, d’encadrer les opérations sous-traitées dans les contrats de prestation de services, de formaliser des politiques de confidentialité des données et des procédures relatives à la gestion des demandes d’exercice des droits.
            </p>
            <p>
              La désignation d’un délégué à la protection des données devient obligatoire pour les organismes et autorités publics, et donc pour les fournisseurs de services.
            </p>
          </section>
          <div
            className="form__group"
          >
            <label
              htmlFor="delegue_protection_donnees"
            >
              Délégué·e à la protection des données
            </label>
            <input
              id="delegue_protection_donnees"
              name="enrollment.delegue_protection_donnees"
              onChange={[Function]}
              type="text"
              value={undefined}
            />
          </div>
          <div
            className="form__group"
          >
            <input
              id="checkbox-cnil"
              name="enrollment.checkbox-cnil"
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-cnil"
            >
              Je déclare avoir accompli mes démarches CNIL en accord avec le règlement général de protection des données
            </label>
          </div>
          <h1
            id="convention"
          >
            Convention
          </h1>
          <section
            className="information-text"
          >
            <p>
              Merci de prendre connaissance de la convention d’adhésion d’API Particulier afin de demander un jeton d’accès en décrivant votre projet.
            </p>
          </section>
          <iframe
            height="800px"
            src="static/docs/charte.pdf"
            width="100%"
          />
          <div
            className="form__group"
          >
            <input
              checked={undefined}
              id="validation_de_convention"
              name="enrollment.validation_de_convention"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="validation_de_convention"
            >
              Je valide la présente convention
            </label>
          </div>
          <div
            className="button-list"
          >
            <button
              className="button"
              id="submit"
              name="subscribe"
              type="submit"
            >
              Soumettre la demande
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <div
      className="container"
    >
      <div
        className="jsx-696190668 footer__logo"
      >
        <a
          className="jsx-696190668"
          href="https://api.gouv.fr/"
        >
          <img
            alt="Logo api.gouv.fr"
            className="jsx-696190668"
            src="/static/images/logo-api.gouv.fr-white.png"
          />
        </a>
        <ul
          className="jsx-696190668 footer__social"
        >
          <li
            className="jsx-696190668"
          >
            <a
              className="jsx-696190668"
              href="https://twitter.com/BetaGouv"
              title="Twitter"
            >
              <img
                alt="Twitter"
                className="jsx-696190668 icon icon-twitter"
                src="/static/images/social/twitter.svg"
              />
            </a>
          </li>
          <li
            className="jsx-696190668"
          >
            <a
              className="jsx-696190668"
              href="https://github.com/betagouv"
              title="Github"
            >
              <img
                alt="Github"
                className="jsx-696190668 icon icon-github"
                src="/static/images/social/github.svg"
              />
            </a>
          </li>
          <li
            className="jsx-696190668"
          >
            <a
              className="jsx-696190668"
              href="mailto:contact@particulier.api.gouv.fr"
              title="Nous écrire un mail"
            >
              <img
                alt="Email"
                className="jsx-696190668 icon icon-mail"
                src="/static/images/social/email.svg"
              />
            </a>
          </li>
        </ul>
      </div>
      <ul
        className="footer__links"
      >
        <li>
          <h2>
            particulier.api.gouv.fr
          </h2>
        </li>
        <li>
          <a
            href="https://status.particulier.api.gouv.fr"
          >
            Status des API
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
